src/stores/Store.ts

```diff
@@ -280,15 +627,86 @@ export default class Store {
 	};
 
 	@action setSelectedSchemaType = (type: SchemaType) => {
+		localStorage.setItem('selectedSchemaType', type);
+		this.indicators = [];
 		this.selectedSchemaType = type;
+		this.prepareForSelectedSchemaType(type);
 		this.setIsSchemaApplied(false);
 	};
 
+	prepareForSelectedSchemaType = (type: SchemaType) => {
+		switch (type) {
+			case 'parsed-message':
+				{
+					const messageList = localStorage.getItem('parsedMessagesHistory')
+						? localStorage.getItem('parsedMessagesHistory')
+						: '';
+					localStorage.removeItem('parsedMessagesHistory');
+					this.parsedMessagesHistory = [];
+					if (messageList !== '') {
```

Comment:
You should avoid type casting whenever its possible.
In this case you should have used
`messageList && typeof messageList === 'string'` so you don't have to cast type `as string`